arviz/plots/dotplot.py

```diff
@@ -0,0 +1,229 @@
+import numpy as np
+
+from ..rcparams import rcParams
+from .plot_utils import get_plotting_function
+
+
+def plot_dot(
+    values=None,
+    binwidth=None,
+    dotsize=1,
+    stackratio=1,
+    hdi_prob=None,
+    rotated=False,
+    dotcolor="C0",
+    intervalcolor="C3",
+    markersize=None,
+    markercolor="C0",
+    marker="o",
+    figsize=None,
+    linewidth=None,
+    point_estimate="auto",
+    nquantiles=50,
+    quartiles=True,
+    point_interval=None,
+    ax=None,
+    show=None,
+    plot_kwargs=None,
+    backend=None,
+    backend_kwargs=None,
+    **kwargs
+):
+
+    """Plot distribution as dot plot or quantile dot plot.
+
+    This function uses the Wilkinson's Algorithm
+    (Leland Wilkinson (1999) Dot Plots, The American Statistician, 53:3, 276-281,
+    DOI: 10.1080/00031305.1999.10474474)to allot dots to bins.
+    The quantile dot plots was inspired from the paper(Matthew Kay, Tara Kola, Jessica R. Hullman,
+    and Sean A. Munson. 2016. When (ish) is My Bus? User-centered Visualizations of Uncertainty in
+    Everyday, Mobile Predictive Systems. DOI:https://doi.org/10.1145/2858036.2858558).
+
+    Parameters
+    ----------
+    values : array-like
+        Values to plot
+    binwidth : float
+        Width of the bin for drawing the dot plot.
+    dotsize : float
+        The size of the dots relative to the bin width. The default, 1, makes dots be
+        just about as wide as the bin width.
+    stackratio : float
+        The distance between the center of the dots in the same stack relative to the bin height.
+        The default, 1, makes dots in the same stack just touch each other.
+    point_interval : bool
+        Plots the point interval. Uses hdi_prob to plot the HDI interval
+    point_estimate : Optional[str]
+        Plot point estimate per variable. Values should be ‘mean’, ‘median’, ‘mode’ or None.
+        Defaults to ‘auto’ i.e. it falls back to default set in rcParams.
+    dotcolor : string
+        The color of the dots
```

Comment:
This should indicate which values are valid (same applies to all kwargs below). I think the best way to do so is by indicating to which matplotlib and bokeh function this is eventually passed to and using intersphinx to link there. Alternatively, if using `to_hex` it can also be simply "must be a valid matplotlib color", because we use a matplotlib helper function to convert the color to hex so all matplotlib colors also work on bokeh. 

Example: https://arviz-devs.github.io/arviz/_modules/arviz/plots/hdiplot.html#plot_hdi (bokeh patch seems to have been deleted/moved and the link is no longer generated correctly)

Guidance on intersphinx: https://arviz-devs.github.io/arviz/contributing/developer_guide.html#hyperlinks